### PR TITLE
[type:feat]Discovery adapts to namespace transformation

### PR DESCRIPTION
**Pull request by xcsnx**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by xcsnx** [comment]
#5623

**Review by moremind — APPROVED** [review]

File: shenyu-admin/src/main/resources/mappers/discovery-sqlmap.xml [hunk 1]
```diff
@@ -165,6 +170,9 @@
             <if test="props != null">
                 #{props, jdbcType=LONGVARCHAR},
             </if>
+            <if test="namespaceId != null">
+                #{namespaceId, jdbcType=VARCHAR},
```
Comment (moremind) [inline shenyu-admin/src/main/resources/mappers/discovery-sqlmap.xml:174]: remove `,`

File: shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/selector/GrpcUpstream.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ private GrpcUpstream(final Builder builder) {
         this.weight = builder.weight;
         setStatus(statusValue);
         setTimestamp(builder.timestamp);
+        setNamespaceId(builder.namespaceId);
```
Comment (moremind) [inline shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/selector/GrpcUpstream.java:48]: why change this? need to change this?@ywj1352

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/selector/GrpcUpstream.java (hunk 2 above)
Comment (ywj1352) [reply shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/selector/GrpcUpstream.java:48]: image
From the functional point of view of the code does not need this at the moment.   But it's not wrong from an integrity point of view

File: shenyu-admin/src/main/resources/mappers/discovery-sqlmap.xml (hunk 1 above)
Comment (xcsnx) [reply shenyu-admin/src/main/resources/mappers/discovery-sqlmap.xml:174]: ok

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/DiscoveryMapper.java [hunk 3]
```diff
@@ -60,9 +60,10 @@ public interface DiscoveryMapper extends ExistProvider {
      *
      * @param pluginName plugin name
      * @param level      level
+     * @param namespaceId namespaceId
      * @return {@linkplain DiscoveryDO}
      */
-    DiscoveryDO selectByPluginNameAndLevel(@Param("pluginName") String pluginName, @Param("level") String level);
+    DiscoveryDO selectByPluginNameAndLevel(@Param("pluginName") String pluginName, @Param("level") String level, @Param("namespaceId") String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/DiscoveryMapper.java:66]: rename the method

**Review by moremind — CHANGES_REQUESTED** [review]

**Review by moremind — APPROVED** [review]

**Pull request merged**